### PR TITLE
fix: editor state initialization

### PR DESCRIPTION
**Pull request by sashuk**
[FX-4129]

### Description

This pull request fixes the editor state initialization.

According to the source code (https://github.com/facebook/lexical/blob/2bcf11bec944cd3bd69b7e7b40903e593b716a06/packages/lexical-react/src/LexicalComposer.tsx#L119), the editor state needs to be initialized either by Lexical itself or inside of the `editorConfig` if it is a function. This pull request removes the case when the function is provided, but the state is not initialized by anybody.
 
### How to test

- Navigate to https://picasso.toptal.net/fx-4129-insert-list-to-empty-editor-2/?path=/story/forms-richtexteditor--richtexteditor

### Screenshots

https://github.com/toptal/picasso/assets/1390758/0a0a72a8-d276-4bb5-a6dd-5a4d89044d0c

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4129]: https://toptal-core.atlassian.net/browse/FX-4129?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/picasso-rich-text-editor/src/LexicalEditor/LexicalEditor.tsx [hunk 1]
```diff
@@ -189,7 +187,6 @@ const LexicalEditor = forwardRef<HTMLDivElement, Props>(function LexicalEditor(
           key={`${disabled || !isFocused}`}
           customEmojis={customEmojis}
           plugins={plugins}
-          testIds={testIds}
```
Comment (sashuk) [inline packages/picasso-rich-text-editor/src/LexicalEditor/LexicalEditor.tsx:192]: Problem from base branch

**Comment by sashuk** [comment]
@toptal-anvil ping reviewers

File: cypress/component/RichTextEditor.spec.tsx [hunk 2]
```diff
@@ -445,5 +445,15 @@ describe('RichTextEditor', () => {
       // the click should not open select but just simply trigger focus on whole editor
       cy.get('@headerSelect').find('input').should('not.have.attr', 'disabled')
     })
+
+    it('creates list from scratch', () => {
```
Comment (TomasSlama) [inline cypress/component/RichTextEditor.spec.tsx:449]: ❤️

Comment (TomasSlama) [reply cypress/component/RichTextEditor.spec.tsx:449]: can you please also remove `skip` from test cases that have `// TODO: https://toptal-core.atlassian.net/browse/FX-4129`

Comment (TomasSlama) [reply cypress/component/RichTextEditor.spec.tsx:449]: if by any chance this would fix even other bug tickets, you could unskip even those 🙇

**Review by dmaklygin — APPROVED** [review]

File: cypress/component/RichTextEditor.spec.tsx (hunk 2 above)
Comment (TomasSlama) [reply cypress/component/RichTextEditor.spec.tsx:449]: also I noticed that `removes header format` cypress test is failing. I double checked that on `feature/rte-lexical` it runs correctly. Could you check why this fix is breaking it? 🤔

File: cypress/component/RichTextEditor.spec.tsx [hunk 3]
```diff
@@ -262,11 +261,22 @@ describe('RichTextEditor', () => {
...
       // remove all
-      cy.get('@editor').type('{selectall}{del}')
+      cy.get('@editor').type(`{selectall}${removalKey}`)
 
       cy.get('@placeholder').should('be.visible')
       buttonShouldNotBeActive(cy.get('@olButton'))
       buttonShouldNotBeActive(cy.get('@boldButton'))
+    }
+
+    describe('using delete key', () => {
```
Comment (sashuk) [inline cypress/component/RichTextEditor.spec.tsx:271]: Delete and backspace behave differently, please see the https://www.computerhope.com/issues/ch001389.htm for details (this way, we need to test both cases)

File: cypress/component/RichTextEditor.spec.tsx (hunk 2 above)
Comment (sashuk) [reply cypress/component/RichTextEditor.spec.tsx:461]: @TomasSlama All related tests were uncommented, the failing test was checked, and additional tests for delete and backspace were added (also the https://toptal-core.atlassian.net/browse/FX-4138 was added to address it later), thank you!

File: cypress/component/RichTextEditor.spec.tsx [hunk 4]
```diff
@@ -440,10 +449,23 @@ describe('RichTextEditor', () => {
         expect(error.message).to.include('prevents user mouse interaction')
       })
 
+      // need for editor to loose focus
+      cy.get('body').click(0, 0)
```
Comment (sashuk) [inline cypress/component/RichTextEditor.spec.tsx:453]: Editor is focused before the beginning of assertions, so need to be focused out again

https://github.com/toptal/picasso/assets/1390758/f00a3b94-9e58-4a40-af43-763ef72ee8f5

File: packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts [hunk 5]
```diff
@@ -10,20 +10,30 @@ import { $generateNodesFromDOM } from '@lexical/html'
...
+
+    lexicalValueNodes.forEach(node => {
+      if ($isElementNode(node) || $isDecoratorNode(node)) {
+        root.append(node)
+      } else {
+        const paragraphNode = $createParagraphNode()
+
+        paragraphNode.append(node)
+        root.append(paragraphNode)
+      }
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts:28]: NIT: wdyt simplifying it to:
```suggestion
      const nodeToAppend = 
        $isElementNode(node) || $isDecoratorNode(node)
          ? node
          : $createParagraphNode().append(node)
          
      root.append(paragraphNode)
```

File: packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts [hunk 6]
```diff
@@ -10,20 +10,30 @@ import { $generateNodesFromDOM } from '@lexical/html'
...
+    })
+  } else {
+    if (root.isEmpty()) {
       const paragraphNode = $createParagraphNode()
 
-      paragraphNode.append(node)
       root.append(paragraphNode)
```
Comment (augustobmoura) [inline packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts:34]: NIT: maybe just join these lines, wdyt?
```ts
      root.append($createParagraphNode())
```

**Review by augustobmoura — APPROVED** [review]
Checked and it works on temploy

Really cool solution @sashuk!
I left a few small NITs that looks could simplify the code, but it can be merged nonetheless

File: packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts (hunk 6 above)
Comment (sashuk) [reply packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts]: @augustobmoura Done, thank you!

File: packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts (hunk 5 above)
Comment (sashuk) [reply packages/picasso-rich-text-editor/src/LexicalEditor/utils/setEditorValue.ts]: @augustobmoura Looks great, added, thanks!

**Comment by dmaklygin** [comment]
it's kinda interesting, but the root node is not empty even after removing the child nodes.
I think, we should investigate the problem deeply

**Comment by sashuk** [comment]
@dmaklygin Do you mean that there is always a `<p/>`, how you detect that it is not empty?

**Comment by dmaklygin** [comment]
> @dmaklygin Do you mean that there is always a `<p/>`, how you detect that it is not empty?

nope, I've checked and this is intentional behavior. 
I confirm my approval!

**Comment by sashuk** [comment]
@dmaklygin Awesome, merging then! Visual are test the only ones that fail

**Pull request merged**